processor/modifyscopeprocessor/factory.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modifyscopeprocessor
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "modifyscope"
+)
+
+func NewFactory() component.ProcessorFactory {
+	return component.NewProcessorFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithMetricsProcessor(createMetricsProcessor, component.StabilityLevelBeta))
+}
+
+func createDefaultConfig() component.Config {
+	settings := config.NewProcessorSettings(component.NewID(typeStr))
+	return &Config{
+		ProcessorSettings: &settings,
+	}
+}
+
+var processorCapabilities = consumer.Capabilities{MutatesData: true}
+
+func createMetricsProcessor(
+	ctx context.Context,
+	params component.ProcessorCreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Metrics,
+) (component.MetricsProcessor, error) {
+	oCfg := cfg.(*Config)
+	if err := validateConfiguration(oCfg); err != nil {
+		return nil, err
+	}
+	metricsProcessor := newModifyScopeProcessor(params.Logger, oCfg)
+	return processorhelper.NewMetricsProcessor(
+		ctx,
+		params,
+		cfg,
+		nextConsumer,
+		metricsProcessor.ProcessMetrics,
+		processorhelper.WithCapabilities(processorCapabilities))
+}
+
+// validateConfiguration validates the input configuration has all of the required fields for the processor
+// An error is returned if there are any invalid inputs.
+func validateConfiguration(config *Config) error {
```

Comment:
[optional] Based on how the Ops Agent is using this processor, are there some formats we want to validate against? Must have some specific prefix for the instrumentation_scope for example

Reply:
Since this isn't for upstreaming or direct user use I don't particularly care if it's hard to use; I'd rather just keep this code as simple as possible here.